From 669e9de926170afcce80b05f65cf1ddfcf6b8bc2 Mon Sep 17 00:00:00 2001
From: Jason Wang <jasowang@redhat.com>
Date: Mon, 29 Sep 2014 00:59:01 -0400
Subject: [virt] hyperv/net: Negotiate suitable ndis version for offload support

Message-id: <1411952358-34104-18-git-send-email-jasowang@redhat.com>
Patchwork-id: 96759
O-Subject: [RHEL7.1 PATCH 17/34] Drivers: net: hyperv: Negotiate suitable ndis version for offload support
Bugzilla: 1146357
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Vitaly Kuznetsov <vkuznets@redhat.com>

From: KY Srinivasan <kys@microsoft.com>

Bugzilla: 1146357

Ws2008R2 supports ndis_version 6.1 and 6.1 is the minimal version required
for various offloads. Negotiate ndis_version 6.1 when on ws2008r2.

Signed-off-by: K. Y. Srinivasan <kys@microsoft.com>
Reviewed-by: Haiyang Zhang <haiyangz@microsoft.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 1f73db495aaa0f1eb4de84856d9ea46711898c08)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/hyperv/netvsc.c b/drivers/net/hyperv/netvsc.c
index 356b80a..d5681bd 100644
--- a/drivers/net/hyperv/netvsc.c
+++ b/drivers/net/hyperv/netvsc.c
@@ -345,7 +345,7 @@ static int netvsc_connect_vsp(struct hv_device *device)
  memset(init_packet, 0, sizeof(struct nvsp_message));
 
  if (net_device->nvsp_version <= NVSP_PROTOCOL_VERSION_4)
-  ndis_version = 0x00050001;
+  ndis_version = 0x00060001;
  else
   ndis_version = 0x0006001e;
 
-- 
1.7.1